wp/lib/bap_wp/src/constraint.ml

```diff
@@ -23,27 +23,11 @@ type z3_expr = Expr.expr
 
 type path = bool Jmp.Map.t
 
-type goal = { goal_name : string; goal_val : z3_expr }
+type reg_map = (z3_expr * z3_expr) list Jmp.Map.t
 
-let goal_to_string (g : goal) : string =
-  Format.sprintf "%s: %s%!" g.goal_name (Expr.to_string (Expr.simplify g.goal_val None))
+type goal = { goal_name : string; goal_val : z3_expr }
```

Comment:
This isn't a comment on this pull request in particular, but goal seems like a slightly odd name for this concept